### PR TITLE
Update documentation of KNX integration

### PR DESCRIPTION
**Pull request by marvin-w**
## Proposed change

Due to changes on how HA uses yaml configuration in the future there has been a major refactoring in the KNX integration. We now no longer define each platform individually under the platform key but instead use the top level integration key.

For the refactoring please use https://github.com/home-assistant/home-assistant.io/pull/14299/commits/a6b4f631418c7339c6205a8365fcfdcf14797b7c for the review.
For the dependency update please use https://github.com/home-assistant/home-assistant.io/pull/14299/commits/55fad298953121c54b314189e1c5502bc0a4f687 for the review.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [x] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/39189
- Link to parent pull request in the Brands repository: -
- This PR fixes or closes issue: -

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by farmio — DISMISSED** [review]
Fine! 👍 thank you!

**Comment by marvin-w** [comment]
@farmio  Please review again: https://deploy-preview-14299--home-assistant-docs.netlify.app/

File: source/_integrations/climate.knx.markdown [hunk 1]
```diff
@@ -54,21 +54,43 @@ to the configuration in this case.:
...
+      operation_mode_standby_address: '5/1/9'
+      min_temp: 7.0
+      max_temp: 32.0
 ```
 
-`operation_mode_frost_protection_address` / `operation_mode_night_address` / `operation_mode_comfort_address` are not necessary if `operation_mode_address` is specified.
+`setpoint_shift_mode` allows the two following DPTs to be used:
+
+- DPT6.002 (default)
+- DPT9.002
```
Comment (farmio) [inline source/_integrations/climate.knx.markdown]:
```suggestion
- "DPT6.002" for 1-byte signed integer (default)
- "DPT9.002" for 2-byte float
```

File: source/_integrations/binary_sensor.knx.markdown [hunk 2]
```diff
@@ -22,11 +22,13 @@ Binary sensors are read-only. To write to the knx-bus configure an exposure [KNX
 
 The `knx` integration must be configured correctly, see [KNX Integration](/integrations/knx).
```
Comment (farmio) [inline source/_integrations/binary_sensor.knx.markdown:23]: I think we can delete this line now. `knx:` is mandatory now anyway.

File: source/_integrations/climate.knx.markdown [hunk 3]
```diff
@@ -106,18 +133,19 @@ target_temperature_state_address:
   required: true
   type: string
 setpoint_shift_address:
-  description: KNX address for setpoint_shift. *DPT 6.010*
+  description: KNX address for setpoint_shift. The DPT used can be configured via setpoint_shift_mode.
```
Comment (farmio) [inline source/_integrations/climate.knx.markdown]:
```suggestion
  description: KNX address for setpoint_shift. The DPT used can be configured via setpoint_shift_mode.  *DPT 6.010 or 9.001 based on setpoint_shift_mode*
```

File: source/_integrations/climate.knx.markdown [hunk 4]
```diff
@@ -106,18 +133,19 @@ target_temperature_state_address:
...
   required: false
   type: string
 setpoint_shift_state_address:
-  description: KNX address for reading setpoint_shift. *DPT 6.010*
+  description: KNX address for reading setpoint_shift. The DPT used can be configured via setpoint_shift_mode.
```
Comment (farmio) [inline source/_integrations/climate.knx.markdown]:
```suggestion
  description: KNX address for reading setpoint_shift. The DPT used can be configured via setpoint_shift_mode.  *DPT 6.010 or 9.001 based on setpoint_shift_mode*
```

File: source/_integrations/climate.knx.markdown [hunk 5]
```diff
@@ -152,6 +180,14 @@ controller_mode_state_address:
   description: KNX address for reading HVAC Control Mode. *DPT 20.105*
   required: false
   type: string
+heat_cool_address:
+  description: KNX address for switching between heat/cool mode.
```
Comment (farmio) [inline source/_integrations/climate.knx.markdown]:
```suggestion
  description: KNX address for switching between heat/cool mode. 0: cooling - 1: heating  *DPT 1.100*
```

File: source/_integrations/climate.knx.markdown [hunk 6]
```diff
@@ -152,6 +180,14 @@ controller_mode_state_address:
...
+  required: false
+  type: string
+heat_cool_state_address:
+  description: KNX address for reading heat/cool mode.
```
Comment (farmio) [inline source/_integrations/climate.knx.markdown]:
```suggestion
  description: KNX address for reading heat/cool mode.  0: cooling - 1: heating  *DPT 1.100*
```

**Review by farmio — COMMENTED** [review]
I have a few suggestions.
Also in cover.knx.markdown
```
invert_position:
  description: Set this to true if your actuator report fully closed as 100%.
```
has been changed. I'd write
```
invert_position:
  description: Set this to true if your actuator report fully closed as 0% in KNX.
```
same for angle.

File: source/_integrations/binary_sensor.knx.markdown (hunk 2 above)
Comment (marvin-w) [reply source/_integrations/binary_sensor.knx.markdown:23]: You still need to define the connection config correctly in order for it to work.

Comment (farmio) [reply source/_integrations/binary_sensor.knx.markdown:23]: in most cases automatic GatewayScanner should work fine 😉

**Review by farmio — APPROVED** [review]

**Comment by marvin-w** [comment]
I took care of all the review comments.

**Review by frenck — APPROVED** [review]
Thanks, @marvin-w 👍

**Pull request merged**